packages/yoroi-extension/app/api/ergo/lib/state-fetch/remoteFetcher.js

```diff
@@ -60,218 +135,151 @@ export class RemoteFetcher implements IFetcher {
-        throw new GetUtxosSumsForAddressesApiError();
-      });
+    return this.axiosPost(`${BackendService}/api/txs/utxoSumForAddresses`, {
+      data: { addresses: body.addresses.map(addr => encode(Buffer.from(addr, 'hex'))) },
+      callerName: nameof(this.getUTXOsSumsForAddresses),
+      errorFactory: errClass(GetUtxosSumsForAddressesApiError),
+    });
   }
 
   getTransactionsHistoryForAddresses: HistoryRequest => Promise<HistoryResponse> = (request) => {
     const { network, ...rest } = request;
     const { BackendService } = network.Backend;
     if (BackendService == null) throw new Error(`${nameof(this.getTransactionsHistoryForAddresses)} missing backend url`);
 
+    const self = this;
     return fixHistoryFunc(async body => {
-      return axios(
-        `${BackendService}/api/v2/txs/history`,
-        {
-          method: 'post',
-          timeout: 2 * CONFIG.app.walletRefreshInterval,
-          data: {
-            ...rest,
-            addresses: body.addresses,
-          },
-          headers: {
-            'yoroi-version': this.getLastLaunchVersion(),
-            'yoroi-locale': this.getCurrentLocale()
-          }
-        }
-      ).then(response => {
-        const data: HistoryResponse = response.data;
-        for (const datum of data) {
-          for (let i = 0; i < datum.inputs.length; i++) {
-            // TODO: remove this once this ticket is merged
-            // https://github.com/ergoplatform/explorer-backend/issues/92
-            if (datum.inputs[i].assets == null) {
-              datum.inputs[i].assets = [];
-            }
-          }
-        }
-        return data;
-      })
-        .catch((error) => {
-          Logger.error(`${nameof(RemoteFetcher)}::${nameof(this.getTransactionsHistoryForAddresses)} error: ` + stringifyError(error));
+      return await self.axiosPost(`${BackendService}/api/v2/txs/history`, {
+        data: { ...rest, addresses: body.addresses },
+        callerName: nameof(this.getTransactionsHistoryForAddresses),
+        errorFactory: error => {
           const errorMessage = error.response.data.error;
           if (
             errorMessage === 'REFERENCE_BLOCK_MISMATCH' ||
             errorMessage === 'REFERENCE_TX_NOT_FOUND' ||
             errorMessage === 'REFERENCE_BEST_BLOCK_MISMATCH'
           ) {
-            throw new RollbackApiError();
+            return new RollbackApiError();
           }
-          throw new GetTxHistoryForAddressesApiError();
-        });
+          return new GetTxHistoryForAddressesApiError();
+        },
+        responseMapper: (data: HistoryResponse) => {
+          for (const datum of data) {
+            datum.outputs.forEach(o => fixUtxoToStringValues(o));
+            // TODO: the inputs fix might potentially be removed
+            // https://github.com/ergoplatform/explorer-backend/issues/92
+            datum.inputs.forEach(i => { i.assets = i.assets ?? [] })
```

Comment:
Just to confirm we don't need to apply `fixUtxoToStringValues` to `inputs` ?

Reply:
Good point! I have missed that returned inputs contain values and assets as well, thought they should be only references to utxos.

Adding new commit for this.